### PR TITLE
Fixing CI failures for charts with dependencies

### LINKED ISSUE #2896: [BUG] lint.sh breaks on charts with dependencies
Please see this CircleCI run: https://circleci.com/gh/kubernetes/charts/1156

The chart (chartmuseum) has a dependency on the common chart, and lint no likey:
```
Runing helm lint on the chart at incubator/chartmuseum
==> Linting incubator/chartmuseum
[ERROR] templates/: render error in "chartmuseum/templates/service.yaml": template: chartmuseum/templates/service.yaml:4:11: executing "chartmuseum/templates/service.yaml" at <include "common.full...>: error calling include: template: no template "common.fullname" associated with template "gotpl"

Error: 1 chart(s) linted, 1 chart(s) failed
```

I fixed this problem a while ago, but it appears that the CI scripts run have changed since?
https://github.com/kubernetes/charts/pull/2313

### PR DESCRIPTION
**Pull request by mattfarina**
Closes #2896

/cc @unguiculus @jdolitsky @viglesiasce

### CODE REVIEW

File: test/circle/install.sh [hunk 1]
```diff
@@ -22,6 +22,12 @@ sudo mv linux-amd64/helm /usr/local/bin
 rm -f helm-${HELM_LATEST_VERSION}-linux-amd64.tar.gz
 rm -rf linux-amd64
 
+# Setup Helm so that it will work with helm dep commands. Only the client
+# needs to be setup. In addition, the incubator repo needs to be
+# available for charts that depend on it.
+helm init -c
+helm repo add incubator https://kubernetes-charts-incubator.storage.googleapis.com/
```
Comment (jdolitsky) [inline test/circle/install.sh:29]: Why is this line required? Shouldnt requiements.yaml/.lock just specify an absolute URL? What about stable?

Comment (mattfarina) [reply test/circle/install.sh:29]: You have to add a repo before you can refer to it. Without adding the incubator repo you'd get errors like...

```sh
$ helm dep up
Error: no repository definition for https://kubernetes-charts-incubator.storage.googleapis.com/. Try 'helm repo add'
```

The stable repo is automatically added by default when `helm init` is run.

Comment (jdolitsky) [reply test/circle/install.sh:29]: Ok I see. Change looks good to me

**Pull request merged**